pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -106,141 +145,273 @@ func (c *Compactor) stopping(_ error) error {
+				From:        chk.From(),
+				Through:     chk.Through(),
+				Checksum:    chk.Checksum,
+			},
+		})
+	}
 
-// part1: Create a compact method that assumes no block/meta files exists (eg first compaction)
-// part2: Write logic to check first for existing block/meta files and does above.
-func (c *Compactor) compactNewChunks(ctx context.Context, dst string) (err error) {
-	// part1
-	series := NoopGetSeries()
-	data := NoopGetChunks()
+	return chunkRefs
+}
 
-	bloom := v1.Bloom{ScalableBloomFilter: *filter.NewDefaultScalableBloomFilter(0.01)}
-	// create bloom filters from that.
-	bloom.Add([]byte(fmt.Sprint(data)))
+// TODO Revisit this step once v1/bloom lib updated to combine blooms in the same series
+func buildBloomBlock(bloomForChks v1.SeriesWithBloom, series Series) (bloomshipper.Block, error) {
+	localDst := createLocalFileName(series)
 
-	// block and seriesList
-	seriesList := []v1.SeriesWithBloom{
-		{
-			Series: series,
-			Bloom:  &bloom,
-		},
+	//write bloom to a local dir
+	builder, err := v1.NewBlockBuilder(v1.NewBlockOptions(), v1.NewDirectoryBlockWriter(localDst))
+	if err != nil {
+		level.Info(util_log.Logger).Log("creating builder", err)
+		return bloomshipper.Block{}, err
 	}
 
-	writer := v1.NewDirectoryBlockWriter(dst)
+	err = builder.BuildFrom(v1.NewSliceIter([]v1.SeriesWithBloom{bloomForChks}))
+	if err != nil {
+		level.Info(util_log.Logger).Log("writing bloom", err)
+		return bloomshipper.Block{}, err
+	}
 
-	builder, err := v1.NewBlockBuilder(
-		v1.BlockOptions{
-			SeriesPageSize: 100,
-			BloomPageSize:  10 << 10,
-		}, writer)
+	blockFile, err := os.Open(filepath.Join(localDst, "bloom"))
 	if err != nil {
-		return err
+		level.Info(util_log.Logger).Log("reading bloomBlock", err)
 	}
-	// BuildFrom closes itself
-	err = builder.BuildFrom(v1.NewSliceIter[v1.SeriesWithBloom](seriesList))
+
+	// read the checksum
+	blockFile.Seek(-4, 2)
+	checksum := make([]byte, 4)
+	blockFile.Read(checksum)
+
+	// Reset back to beginning
+	blockFile.Seek(0, 0)
+
+	objectStoreDst := createObjStorageFileName(series, fmt.Sprint(binary.BigEndian.Uint32(checksum)))
+
+	blocks := bloomshipper.Block{
+		BlockRef: bloomshipper.BlockRef{
+			Ref: bloomshipper.Ref{
+				TenantID:       series.tenant,
+				TableName:      series.tableName,
+				MinFingerprint: uint64(series.fingerPrint), //TODO will change once we compact multiple blooms into a block
+				MaxFingerprint: uint64(series.fingerPrint),
+				StartTimestamp: series.from.Unix(),
+				EndTimestamp:   series.through.Unix(),
+				Checksum:       binary.BigEndian.Uint32(checksum),
+			},
+			IndexPath: series.indexPath, // For reviewer, this is the same filepath with the original TSDBindex for this series?
+			BlockPath: objectStoreDst,
+		},
+		Data: blockFile,
+	}
+
+	return blocks, nil
+}
+
+func listSeries(ctx context.Context, objectClient client.ObjectClient) ([]Series, error) {
+	// Returns all the TSDB files, including subdirectories
+	prefix := "index/"
```

Comment:
Heads up, this prefix is [configurable](https://github.com/grafana/loki/blob/main/pkg/storage/config/schema_config.go#L482) and should be factored into inputs or handled elsewhere.

Reply:
This `listSeries` method was my temporary hack to get tableNames and tenants,  i'm aware that it's partial and fragile.
 
[Salva's ring work here](https://github.com/grafana/loki/compare/salvacorts/bloomcompactor-sharding) provides them nicely, it uses `indexShipper/storage`'s `ListTables` [method](https://github.com/grafana/loki/blob/main/pkg/storage/stores/shipper/indexshipper/storage/client.go#L65-L77)

 `Job` struct in Salva's branch will replace the `Series` struct in this PR.